ci-setups/php-gitlab.md

```diff
@@ -0,0 +1,140 @@
+
+2. A new page will open for editing and creating the `.gitlab-ci.yml` file. Remove all the content that exists in the file, and paste the following code in its place:
+
+    ```yaml
+    stages:
+      - lint
+
+    lint:
+      stage: lint
+      tags:
+        - PHP-8.1
+      rules:
+        # Run pipeline if a commit is made to these branches
+        - if: $CI_COMMIT_BRANCH == "main"
+        # Run pipeline if a merge request is created for these branches
+        - if: $CI_MERGE_REQUEST_TARGET_BRANCH_NAME == "main"
+      script:
+        - echo "Copy .env file"
+        - cp ${ENV_FILE} .env
+        - echo "Updating dependencies"
+        - composer update
+        - echo "Installing dependencies"
+        - composer install --prefer-dist --no-progress
+        - echo "Linting code"
+        - composer run lint
+        - echo "Running static code analysis"
+        - composer run phan
+    ```
+
+    Please note that depending on the needs of the projects, the following things can be reconfigured in the above content:
+
+    - ```yaml
+      tags:
+        - PHP-8.1
+      ```
+
+      This value specifies which runner to use for running this pipeline job. It is possible that the value of this may need to be updated in case of multiple runners.
+
+    - ```yaml
+      rules:
+        # Run pipeline if a commit is made to these branches
+        - if: $CI_COMMIT_BRANCH == "main"
+        # Run pipeline if a merge request is created for these branches
+        - if: $CI_MERGE_REQUEST_TARGET_BRANCH_NAME == "main"
+      ```
+
+      This section adds the rules for when the job should run. In its current state, this job will run only when the commit branch is main (i.e., a commit is made to the main branch), or a merge request is created for the main branch.
+
+      In order to add more branches, it can done as follows:
+
+      ```yaml
+      rules:
+        # Run pipeline if a commit is made to these branches
+        - if: $CI_COMMIT_BRANCH == "main" || $CI_COMMIT_BRANCH == "some-branch" || $CI_COMMIT_BRANCH == "some-other-branch"
+        # Run pipeline if a merge request is created for these branches
+        - if: $CI_MERGE_REQUEST_TARGET_BRANCH_NAME == "main" || $CI_MERGE_REQUEST_TARGET_BRANCH_NAME == "another-branch"
+      ```
+
+      This is evaluated as follows:
+      - If the commit branch is 'main' OR 'some-branch' OR 'some-other-branch', run the job.
+      - If a merge request is created and target branch is 'main' OR 'another-branch', run the job.
+
+    - ```yaml
+      - echo "Copy .env file"
+      - cp ${ENV_FILE} .env
+      ```
+
+      This can be removed if the project does not have or require any `.env` file for composer operations.
+
+      > Note: Here, the CI/CD Variable `ENV_FILE` is of type `FILE`. Depending on the type of Variable being used, this section can be edited as required.
+
+    - ```yaml
+      - echo "Updating dependencies"
+      - composer update
```

Comment:
I would not do this.

If we want to update the dependencies then it has to be updated in local and tested first.

But if in local I don't update and also on server where I deploy code I don't update then no point of updating and checking in CI.

Better to remove.

Reply:
This has been removed in the latest commit along with the step creation of .env file.